src/routes/CreateSafePage/steps/NameNewSafeStep.tsx

```diff
@@ -28,9 +35,42 @@ function NameNewSafeStep(): ReactElement {
   }, [provider, setCurrentStep])
 
   const createNewSafeForm = useForm()
-
   const formValues = createNewSafeForm.getState().values
 
+  useEffect(() => {
+    const getInitialOwnerENSNames = async () => {
+      const formValues = createNewSafeForm.getState().values
+      const owners = formValues[FIELD_SAFE_OWNERS_LIST]
+      const ownersWithENSName = await Promise.all(
+        owners.map(async ({ addressFieldName }) => {
+          const address = formValues[addressFieldName]
+          const ensName = await reverseENSLookup(address)
+          const ensDomain = removeTld(ensName)
+          return {
+            address,
+            name: ensDomain,
+          }
+        }),
+      )
+
+      const ownersWithENSNameRecord = ownersWithENSName.reduce<Record<string, string>>((acc, { address, name }) => {
+        return {
+          ...acc,
+          [address]: name,
+        }
+      }, {})
+
+      setOwnersWithENSName(ownersWithENSNameRecord)
+    }
+    getInitialOwnerENSNames()
+  }, [createNewSafeForm])
+
+  useEffect(() => {
+    if (ownersWithENSName) {
+      createNewSafeForm.change(FIELD_SAFE_OWNER_ENS_LIST, ownersWithENSName)
+    }
+  }, [ownersWithENSName, createNewSafeForm])
```

Comment:
Does it work for subsequent form value changes?
In my understanding `createNewSafeForm` will refer to the same object even though values inside might change. So it probably won't trigger the effect.

Reply:
We only want to fetch ENS once for the initial owners. Before it was in `CreateSafePage` but the async caused issues with setting the other initial values so I moved it to the first step. If values change later on (adding new owners) they are handled in the next step `OwnersAndConfirmationsNewSafeStep`. Our Linter complains if we don't include `createNewSafeForm` in the dependency array even though it has no effect.